src/Bridge/NelmioApiDoc/Parser/ApiPlatformParser.php

```diff
@@ -0,0 +1,209 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ApiPlatform\Core\Bridge\NelmioApiDoc\Parser;
+
+use ApiPlatform\Core\Metadata\Property\Factory\PropertyMetadataFactoryInterface;
+use ApiPlatform\Core\Metadata\Property\Factory\PropertyNameCollectionFactoryInterface;
+use ApiPlatform\Core\Metadata\Property\PropertyMetadata;
+use ApiPlatform\Core\Metadata\Resource\Factory\ResourceMetadataFactoryInterface;
+use ApiPlatform\Core\Metadata\Resource\Factory\ResourceNameCollectionFactoryInterface;
+use ApiPlatform\Core\Metadata\Resource\ResourceMetadata;
+use Nelmio\ApiDocBundle\DataTypes;
+use Nelmio\ApiDocBundle\Parser\ParserInterface;
+use Symfony\Component\PropertyInfo\Type;
+
+/**
+ * Extract input and output information for the NelmioApiDocBundle.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+class ApiPlatformParser implements ParserInterface
+{
+    const IN_PREFIX = 'api_platform_in';
+    const OUT_PREFIX = 'api_platform_out';
+    const IRI = 'IRI';
+
+    private static $typeMap = [
+        Type::BUILTIN_TYPE_BOOL => DataTypes::BOOLEAN,
+        Type::BUILTIN_TYPE_FLOAT => DataTypes::FLOAT,
+        Type::BUILTIN_TYPE_INT => DataTypes::INTEGER,
+        Type::BUILTIN_TYPE_STRING => DataTypes::STRING,
+    ];
+
+    private $resourceNameCollectionFactory;
+    private $resourceMetadataFactory;
+    private $propertyNameCollectionFactory;
+    private $propertyMetadataFactory;
+
+    public function __construct(
+        ResourceNameCollectionFactoryInterface $resourceNameCollectionFactory,
+        ResourceMetadataFactoryInterface $resourceMetadataFactory,
+        PropertyNameCollectionFactoryInterface $propertyNameCollectionFactory,
+        PropertyMetadataFactoryInterface $propertyMetadataFactory
+    ) {
+        $this->resourceNameCollectionFactory = $resourceNameCollectionFactory;
+        $this->resourceMetadataFactory = $resourceMetadataFactory;
+        $this->propertyNameCollectionFactory = $propertyNameCollectionFactory;
+        $this->propertyMetadataFactory = $propertyMetadataFactory;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supports(array $item) : bool
+    {
+        $data = explode(':', $item['class'], 2);
+        if (isset($data[1])) {
+            foreach ($this->resourceNameCollectionFactory->create() as $resourceClass) {
```

Comment:
Anyway if the method return `true` we will create it later. Metadata will be cached (I'll add a PSR-6 factory, I just wait for Symfony to finish to integrate the new Cache component), so it will be build in the `supports` method, cached, and retrieved from the cache directly in the `parse` method.